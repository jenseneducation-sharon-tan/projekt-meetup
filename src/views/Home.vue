<template>
  <div class="home">
    <div class="top"></div>
    <div class="intro">
      <Header :mainpage="true" />
    </div>
    <img src="../assets/meetup.jpg" class="meet" />
    <h2 id="event">{{ upcomingEvents }}</h2>
    <EventList :events="events" />
    <h2 id="history">{{ pastEvents }}</h2>
    <EventList :events="eventHistory" :review="true" />
    <Footer />
  </div>
</template>

<script>
import Header from "../components/Header";
import EventList from "../components/EventList";
import Footer from "../components/Footer";
import { mapGetters, mapActions } from "vuex";

export default {
  components: {
    Header,
    EventList,
    Footer,
  },
  data: () => {
    return {
      upcomingEvents: "Upcoming Events",
      pastEvents: "Past Events",
    };
  },
  computed: mapGetters(["events", "eventHistory"]),
  methods: {
    ...mapActions(["fetchEventList", "fetchPastEvents"]),
  },
  created() {
    this.fetchEventList();
    this.fetchPastEvents();
  },
};
</script>

<style scoped>
@import url("https://fonts.googleapis.com/css2?family=Permanent+Marker&display=swap");
.top {
  height: 60px;
  background: rgb(2, 0, 36);
  background: linear-gradient(
    243deg,
    rgba(2, 0, 36, 1) 0%,
    rgba(221, 80, 168, 1) 100%,
    rgba(0, 212, 255, 1) 100%
  );
}
h2 {
  margin: auto 0;
  font-size: 2rem;
  padding: 20px 30px;
}
.meet {
  width: 100%;
}
</style>
