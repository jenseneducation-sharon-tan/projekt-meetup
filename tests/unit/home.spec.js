import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Home from "@/views/Home.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("Home", () => {
  /*  it('should render page when go to "/" ', () => {
    //Arrange
    const wrapper = shallowMount(Home, { localVue });
    //Assert
    expect(wrapper.element).toMatchSnapshot();
  }); */
  test('should render page when go to "/" ', () => {
    //Arrange
    const wrapper = shallowMount(Home, { localVue });
    //Assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
