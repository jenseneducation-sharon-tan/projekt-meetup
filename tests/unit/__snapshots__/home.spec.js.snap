// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render page when go to "/"  1`] = `
<div
  class="home"
>
  <div
    class="top"
  />
   
  <div
    class="intro"
  >
    <div
      class="catchphrase"
    >
      <a>
        <img
          class="logo"
          src="../assets/hangout.jpg"
        />
      </a>
       
      <h2>
        Passionate about something? Let's meetup !
      </h2>
       
      <div
        class="topbar-responsive-links"
      >
        <div
          class="top-bar-right"
        >
          <ul
            class="menu"
          >
            <li>
              <a
                href="#event"
              >
                Current Events
              </a>
            </li>
             
            <li>
              <a
                href="#history"
              >
                Past Events
              </a>
            </li>
          </ul>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <img
    class="meet"
    src="../assets/meetup.jpg"
  />
   
  <h2
    id="event"
  >
    Upcoming Events
  </h2>
   
  <div
    class="event-list-container"
  >
    <div
      class="list-container"
    >
      <div
        class="going"
      />
    </div>
  </div>
   
  <h2
    id="history"
  >
    Past Events
  </h2>
   
  <div
    class="event-list-container"
  >
    <div
      class="list-container"
    >
      <div
        class="review"
      />
    </div>
  </div>
   
  <div
    class="footer"
  >
    <h2>
      Contact us
    </h2>
     
    <div
      class="contact"
    >
      <div
        class="mail"
      >
        <a
          href="mailto:mailus@meetup.com"
        >
          mailus@meetup.com
        </a>
      </div>
       
      <div
        class="call"
      >
        <a>
          072-99 9999
        </a>
      </div>
    </div>
     
    <span>
      All rights reserved @ meetup.se
    </span>
  </div>
</div>
`;
